### PR TITLE
Added a workflow for pre-commit update

### LINKED ISSUE #479: Task: Implement GH workflow for automated pre-commit hook update
### What is the task about?

Following the example of https://github.com/repository-service-tuf/repository-service-tuf-cli/blob/main/.github/workflows/update-pre-commit-hooks.yml, add a GitHub workflow that automates the updating of the `pre-commit` hooks in the `.pre-commit-config.yaml` file.

### Parent feature

_No response_

### References

- https://docs.github.com/en/actions/using-workflows
- https://github.com/repository-service-tuf/repository-service-tuf-cli/blob/main/.github/workflows/update-pre-commit-hooks.yml
- https://github.com/repository-service-tuf/repository-service-tuf-cli/pull/383

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by muhahahmad68**
# Description

Closes #479

# Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

# Additional requirements
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

# Code of Conduct
By submitting this PR, you agree to follow our [Code of Conduct](CODE_OF_CONDUCT.rst).

- [x] I agree to follow this project's Code of Conduct

### CODE REVIEW

**Review by KAUTH — CHANGES_REQUESTED** [review]
Hey @muhahahmad68, thanks for your contribution!
Before approving this PR, let me try to explain a bit more the changes we are trying to achieve in the relevant issue.
- `pre-commit` is a tool we use to manage git commit hooks for everyone working on the project. We also use them in CI with tools such as `bandit` to check our code.
- The `.pre-commit-config.yaml` is where we keep the configuration of this tool and the versions of the hooks we use
- We want to automate the update of the hooks by creating a GitHub workflow which runs periodically.

I would suggest taking a look at https://github.com/repository-service-tuf/repository-service-tuf-cli/blob/main/.github/workflows/update-pre-commit-hooks.yml to get a better understanding of the changes we need.

**Comment by muhahahmad68** [comment]
Kindly some clarifications, @KAUTH 
1. Is it meant to be included in the file like I did?
2. To an extent I know what pre-commit does, can you be more specific on the changes I need to make?

I went through the link you sent and read up on some documentation/online resources before writing the code. I guess I'm still missing some crucial things. Bear with me as this is my first time writing a workflow.

**Comment by KAUTH** [comment]
> Kindly some clarifications, @KAUTH
> 
> 1. Is it meant to be included in the file like I did?
> 2. To an extent I know what pre-commit does, can you be more specific on the changes I need to make?
> 
> I went through the link you sent and read up on some documentation/online resources before writing the code. I guess I'm still missing some crucial things. Bear with me as this is my first time writing a workflow.

Hey @muhahahmad68 no worries, I can explain a bit more.

What we need to do for this task is copy exactly the contents of [this](https://github.com/repository-service-tuf/repository-service-tuf-cli/blob/main/.github/workflows/update-pre-commit-hooks.yml) file and add it to a new file which you have to create, `.github/workflows/update-pre-commit-hooks.yml`.

To explain what this GH workflow does shortly:
This GH workflow automatically updates pre-commit hooks, running on a daily schedule. It checks out the code, sets up Python, installs necessary dependencies, and updates the hooks. If changes are detected, it runs tests to ensure stability and then creates a pull request with the updates, applying specific labels and commit messages.

The file you modified (`.pre-commit-config.yaml`) is where we keep the configuration of the `pre-commit` tool and the versions of the hooks we use and it's actually the file that will be automatically updated by the process explained above.

If you want to understand in more depth how the GH workflow works I would suggest taking it step by step and following the [GH docs](https://docs.github.com/en/actions/using-workflows).

**Comment by muhahahmad68** [comment]
Really appreciate, @KAUTH 
Done as you've said, and I'll make sure to go through the link you sent

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -40,3 +40,4 @@ repos:
       files: ''
       pass_filenames: false
       verbose: true
+
```
Comment (KAUTH) [inline .pre-commit-config.yaml:43]: One small comment here, please remove the newline.

**Review by KAUTH — CHANGES_REQUESTED** [review]
Very nice!
Just a small thing we should fix before we merge this.

**Review by KAUTH — APPROVED** [review]
LGTM

**Pull request merged**